src/libs/iOSSafariAutoScrollback/index.web.js

```diff
@@ -0,0 +1,87 @@
+/* The autoScrollBack address Mobile Safari-specific issues when the user overscrolls the window while the keyboard is visible */
+import Str from 'expensify-common/lib/str';
+import smoothscrollPolyfill from 'smoothscroll-polyfill';
+import CONST from '../../CONST';
+import getBrowser from '../getBrowser';
+
+const userAgent = navigator.userAgent.toLowerCase();
+
+// The innerHeight when the keyboard is not visible
+const baseInnerHeight = window.innerHeight;
+
+// Control flag if an input/text area was focused and we're waiting the "focus scroll" to identify the screen size
+let isWaitingForScroll = false;
+
+// Calculated value of the maximum value to the screen to scroll when keyboard is visible
+let maxScrollY = 0;
+
+let isTouching = false;
+
+let scrollbackTimeout;
+
+const isIOS15 = Str.contains(userAgent, 'iphone os 15_');
+
+function scrollback() {
+    if (isTouching || maxScrollY >= window.scrollY) {
+        return;
+    }
+    window.scrollTo({top: maxScrollY, behavior: 'smooth'});
+}
+
+function scheduleScrollback() {
+    if (!maxScrollY) {
+        return;
+    }
+
+    if (scrollbackTimeout) {
+        clearTimeout(scrollbackTimeout);
+        scrollbackTimeout = undefined;
+    }
+
+    if (!isTouching && window.scrollY > maxScrollY) {
+        scrollbackTimeout = setTimeout(scrollback, 34);
+    }
+}
+
+
+function touchStarted() {
+    isTouching = true;
+}
+
+function scrollbackAfterTouch() {
+    isTouching = false;
+    scrollback();
+}
+
+function scrollbackAfterScroll() {
+    if (isWaitingForScroll && !maxScrollY) {
+        isWaitingForScroll = false;
+        const keyboardHeight = baseInnerHeight - window.visualViewport.height;
+
+        // The iOS 15 Safari has a 52 pixel tall address label that must be manually added
+        maxScrollY = keyboardHeight + (isIOS15 ? 52 : 0);
+    }
+
+    scheduleScrollback();
+}
+
+function startWaitingForScroll() {
+    isWaitingForScroll = true;
+}
+
+function stopWaitingForScroll() {
+    isWaitingForScroll = false;
+}
+
+
+export default function () {
+    if (!getBrowser() === CONST.BROWSER.SAFARI || !Str.contains(userAgent, 'iphone os 1')) {
```

Comment:
Isn't this supposed to be 
```js
getBrowser() !== CONST.BROWSER.SAFARI
```

`!getBrowser()` would be evaluated to `boolean` otherwise, so it's most probably the second part of the check that is doing all the work here

Reply:
WOW. It should be getBrowser() == CONST.BROWSER.SAFARI

Reply:
So we want to exist early if it regular safari, and only apply the polyfill for mobile ?

Reply:
This is an exist early condition 
I though I tried to exist early if the browser is not Safari or mobile safari

Reply:
IT should only work on Safari web.

Reply:
OMG 🤦‍♂️

Reply:
@kidroca if possible, load that file only for Website + Safari

Reply:
Hey I'm slightly confused I thought the issue is "Mobile Web Safari: Adds auto scroll back" 
Which mean we're only targeting ios Safari (iPhone & iPads) and should apply the fix there, correct?

My plan is to use this expression 

```js
if (_.every([/iP(ad|hone)/i, /WebKit/i, /CriOS/i], re => re.test(navigator.userAgent))) {
   const applyAutoScrollbackFix = require('../../libs/iOSSafariAutoScrollback');
   applyAutoScrollbackFix();
}
```

The check is based on: https://stackoverflow.com/a/29696509/4834103

I can extract 

```js
const isMobileSafari = () => _.every([/iP(ad|hone)/i, /WebKit/i, /CriOS/i], re => re.test(navigator.userAgent));
```

But I don't know where to put it
Since it's only one time usage so far, I think I'll just keep it inline 
Otherwise `getBrowser` seems to be a nice place, but I'm not sure how exactly to fit this expression there, or just add a `isMobileSafari` check